[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion

ZhiweiYan-96 opened the pull request:
# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118064
* __->__ #117824
* #117529
* #117512



cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

ZhiweiYan-96 committed:
Intel GPU oneDNN upstreaming: Linear pointwise fusion

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

gujinghui commented on aten/src/ATen/native/mkldnn/xpu/Linear.cpp:
```diff
@@ -0,0 +1,100 @@
+namespace native::xpu {
+
+using namespace impl;
+#define IPEX_LINEAR_DEFINATION(func)                                       \
```
pls remove IPEX prefix.

ZhiweiYan-96 replied:
Has removed, thanks~

gujinghui commented on aten/src/ATen/native/mkldnn/xpu/Linear.cpp:
```diff
@@ -0,0 +1,100 @@
+    return linear_wrapper.call(result, input, weight, bias, post_op);      \
+  }
+
+#define IPEX_LINEAR_BINARY_DEFINATION(func)                                \
```
ditto.

ZhiweiYan-96 replied:
removed

xiaolil1 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

xiaolil1 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

EikanWang commented on aten/src/ATen/native/mkldnn/xpu/FusionUtils.cpp:
```diff
@@ -0,0 +1,106 @@
+
+using namespace at::native::xpu::onednn;
+
+namespace at {
+namespace native::xpu {
```
Refine the namespace.

ZhiweiYan-96 replied:
refined

EikanWang commented on aten/src/ATen/native/mkldnn/xpu/FusionUtils.h:
```diff
@@ -0,0 +1,27 @@
+
+namespace at {
+namespace native::xpu {
+xpu::onednn::Attr unary_attr_with_arg(
```
Since it is under `xpu` namespace, `xpu::` should be redundant.

ZhiweiYan-96 replied:
has changed to `at::native::onednn`

EikanWang commented:
Please add test cases.

leizhenyuan committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

gujinghui commented on aten/src/ATen/native/mkldnn/xpu/FusionUtils.cpp:
```diff
@@ -0,0 +1,104 @@
+
+namespace at::native::xpu {
+
+#define ATTR_FUNC(NAME)                                                       \
```
Do we really need this to make code unreadable?

ZhiweiYan-96 replied:
Thanks for the suggestion! I would remove such marco. Agree that this would hinder the readability to user and its unneeded to add a maro this small utils.

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

jgong5 reviewed (commented):
How are you going to test your PR.

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Linear pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117202 and its further predecessor PRs. i

This PR is intended to add post-op fusion support fo Linear. The liner-pointwise fusion is expected to be used in graph mode like `torch.compile`. The `FusionUtils.cpp` file defines a utilization APIs for generating primitive attribute. This APIs would also be used for conv-pointwise fusion, which is in #118064.





cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]
